### PR TITLE
[rebranch] Various changes to get rebranch compiling after taking new clang branch

### PR DESCRIPTION
**Pull request by bnbarham**

### CODE REVIEW

**Comment by bnbarham** [comment]
@egorzhdan and @beccadax for clang importer changes.

@aschwaighofer for IRGen/SIL.

**Comment by bnbarham** [comment]
https://github.com/apple/llvm-project/pull/8938

@swift-ci please test

**Comment by bnbarham** [comment]
https://github.com/apple/llvm-project/pull/8938

@swift-ci please test

**Comment by bnbarham** [comment]
https://github.com/apple/llvm-project/pull/8938

@swift-ci please test

**Comment by bnbarham** [comment]
https://github.com/apple/llvm-project/pull/8938

@swift-ci please test

**Review by egorzhdan — APPROVED** [review]
ClangImporter changes LGTM

**Review by aschwaighofer — APPROVED** [review]

**Review by compnerd — APPROVED** [review]

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -1510,19 +1510,17 @@ ClangImporter::create(ASTContext &ctx,
   importer->Impl.objectAtIndexedSubscript
     = clangContext.Selectors.getUnarySelector(
         &clangContext.Idents.get("objectAtIndexedSubscript"));
-  clang::IdentifierInfo *setObjectAtIndexedSubscriptIdents[2] = {
-    &clangContext.Idents.get("setObject"),
-    &clangContext.Idents.get("atIndexedSubscript")
-  };
+  const clang::IdentifierInfo *setObjectAtIndexedSubscriptIdents[2] = {
+      &clangContext.Idents.get("setObject"),
+      &clangContext.Idents.get("atIndexedSubscript")};
```
[the remark this answers is not in the document]
Comment (bnbarham) [reply lib/ClangImporter/ClangImporter.cpp:1515]: `git clang-format` 🤷 I didn't do any manual formatting. Are you not seeing the same?

File: lib/ClangImporter/ClangImporter.cpp [hunk 2]
```diff
@@ -1510,19 +1510,17 @@ ClangImporter::create(ASTContext &ctx,
...
   importer->Impl.objectForKeyedSubscript
     = clangContext.Selectors.getUnarySelector(
         &clangContext.Idents.get("objectForKeyedSubscript"));
-  clang::IdentifierInfo *setObjectForKeyedSubscriptIdents[2] = {
-    &clangContext.Idents.get("setObject"),
-    &clangContext.Idents.get("forKeyedSubscript")
-  };
+  const clang::IdentifierInfo *setObjectForKeyedSubscriptIdents[2] = {
+      &clangContext.Idents.get("setObject"),
+      &clangContext.Idents.get("forKeyedSubscript")};
```
[the remark this answers is not in the document]
Comment (bnbarham) [reply lib/ClangImporter/ClangImporter.cpp:1523]: Same here

File: lib/ClangImporter/ClangImporter.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (compnerd) [reply lib/ClangImporter/ClangImporter.cpp:1515]: trailing comma would've kept the line break .. might be easier to spot the diff that way, but 🤷‍♂️

**Review by compnerd — APPROVED** [review]

File: lib/ClangImporter/ClangImporter.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bnbarham) [reply lib/ClangImporter/ClangImporter.cpp:1515]: I can add it in 👍, I have more changes to go in anyway as apparently CI uses quite old SDKs. Thanks for reviewing!

[the remark this answers is not in the document]
Comment (bnbarham) [reply lib/ClangImporter/ClangImporter.cpp]: Turns out clang-format double indents here anyway, so it's still different formatting 😅.

**Comment by bnbarham** [comment]
https://github.com/apple/llvm-project/pull/8938

@swift-ci please test

**Comment by bnbarham** [comment]
https://github.com/apple/llvm-project/pull/8938

@swift-ci please test

**Comment by bnbarham** [comment]
Heaps more failures than when I checked `next` a couple weeks ago :(. As far as I can tell mostly caused by:
https://github.com/llvm/llvm-project/pull/93873
and
https://github.com/llvm/llvm-project/pull/91724

**Pull request merged**